### ISSUE TITLE
`UnnecessaryFullyQualifiedName` false negative when class defined in the same package

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#9572]
## Expected Behavior
(Probably this is easier to understand reading the tests below)

If I have an android module with namespace `bar` android generates me by default a class `bar.R`. If in a class defined in the package `bar` I call `foo.R` directly (no import) the rule should report it because I could avoid the `foo.R` with an import.

## Observed Behavior
It doesn't because `R` could be `bar.R` and it doesn't need any import. In this case is important to check if `bar.R` is used.

## Steps to Reproduce
I wrote the tests:
```kotlin
        @Test
        fun `does report when a class is defined in the same package but not used`() {
            val code = """
                package foo

                class Asdf

                fun main() {
                    bar.Asdf("")
                }
            """.trimIndent()
            val extraClassCode = """
                package bar
                class Asdf(string: String)
            """.trimIndent()

            assertThat(subject.lintWithContext(env, code, extraClassCode)).hasSize(1)
        }

        @Test
        fun `does not report when a class is defined in the same package and it is used`() {
            val code = """
                package foo

                class Asdf

                fun main() {
                    Asdf()
                    bar.Asdf("")
                }
            """.trimIndent()
            val extraClassCode = """
                package bar
                class Asdf(string: String)
            """.trimIndent()

            assertThat(subject.lintWithContext(env, code, extraClassCode)).isEmpty()
        }
```

## Context
I think that this is a regression added when fixing #9322 (PR #9329)

cc @mirkoalicastro

## Your Environment
* Version of detekt used: 2.0alpha6 & main
